provisioning_templates/provision/kickstart_default.erb

```diff
@@ -73,15 +73,40 @@ selinux --<%= host_param('selinux-mode') || host_param('selinux') || 'enforcing'
 keyboard <%= host_param('keyboard') || 'us' %>
 skipx
 
-<% subnet = @host.subnet -%>
-<% if subnet.respond_to?(:dhcp_boot_mode?) -%>
-<% dhcp = subnet.dhcp_boot_mode? && !@static -%>
+<%
+  # This template only supports either IPv4 or IPv6 but not both
+  if os_major >= 6
+    common_network_options = "--hostname #{@host} --device=#{@host.mac}"
+  else
+    common_network_options = "--hostname #{@host}"
+  end
+  if @host.subnet
+    subnet = @host.subnet
+    if @static
+      dhcp = false
+    elsif subnet.nil?
+      dhcp = true
+    else
+      dhcp = subnet.dhcp_boot_mode?
+    end
+-%>
+network --bootproto <%= dhcp ? 'dhcp' : "static --ip=#{@host.ip} --netmask=#{subnet.mask} --gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu.to_s}" %> <%= common_network_options -%>
```

Comment:
`.to_s` in string interpolation is redundant, right?
```suggestion
network --bootproto <%= dhcp ? 'dhcp' : "static --ip=#{@host.ip} --netmask=#{subnet.mask} --gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu}" %> <%= common_network_options -%>
```

Reply:
Yes, good one.